packages/flutter_tools/lib/src/commands/generate.dart

```diff
@@ -34,12 +33,34 @@ class GenerateCommand extends FlutterCommand {
     codegenDaemon.startBuild();
     await for (CodegenStatus codegenStatus in codegenDaemon.buildResults) {
       if (codegenStatus == CodegenStatus.Failed) {
-        throwToolExit('Code generation failed');
+        printError('Code generation failed.');
+        break;
       }
       if (codegenStatus ==CodegenStatus.Succeeded) {
         break;
       }
     }
+    // Check for errors.
+    final Directory buildDirectory = flutterProject.dartTool.childDirectory('build');
+    final Directory errorCacheParent = buildDirectory.listSync().firstWhere((FileSystemEntity entity) {
+      return entity is Directory && entity.childDirectory('error_cache').existsSync();
+    }, orElse: () => null);
+    if (errorCacheParent == null) {
+      return null;
```

Comment:
Should this return `FlutterCommandResult` with `ExitStatus.success`?

Reply:
null is the same as success